File: Dot examples/ASP Book Examples/Book Examples/Chapter34/cs/App_Code/Entities/Chapter.cs
using System;
using System.Linq;
using System.Collections.Generic;
using System.Data.Linq.Mapping;
using System.Data.Linq;

/// <summary>
/// Summary description for Chapter
/// </summary>
public partial class Chapter : EntityBase<Chapter>
{

    public static IEnumerable<Chapter> SelectByBookId(int bookId)
    {
        return Table.Where( c => c.BookId == bookId ).OrderBy( c => c.OrderNumber);
    }



    protected override void Validate()
    {
        if (!ValidationUtility.SatisfiesRequired(Title))
            ValidationErrors.Add("Title", "Title is required", "required");
        if (!ValidationUtility.SatisfiesRequired(Description))
            ValidationErrors.Add("Description", "Description is required", "required");

    }
    




}

File: Dot examples/ASP Book Examples/Book Examples/Chapter34/vb/App_Code/Entities/Chapter.vb
Imports System
Imports System.Linq
Imports System.Collections.Generic
Imports System.Data.Linq.Mapping
Imports System.Data.Linq

''' <summary>
''' Summary description for Chapter
''' </summary>
Partial Public Class Chapter
     Inherits EntityBase(Of Chapter)

    Public Shared Function SelectByBookId(ByVal bookId As Integer) As IEnumerable(Of Chapter)
        Return Table.Where(Function(c) c.BookId = bookId).OrderBy(Function(c) c.OrderNumber)
    End Function



    Protected Overrides Sub Validate()
        If Not ValidationUtility.SatisfiesRequired(Title) Then
            ValidationErrors.Add("Title", "Title is required", "required")
        End If
        If Not ValidationUtility.SatisfiesRequired(Description) Then
            ValidationErrors.Add("Description", "Description is required", "required")
        End If
    End Sub





End Class

